### PR TITLE
Add support for sending messages to certain teams or players with lua

### PR DESCRIPTION
**Pull request by abcdefg30**
This extends the normal `DisplayMessage` to include a team parameter (0 means to everyone).
And secondly adds a new method for sending private messages which accepts a player parameter.

### CODE REVIEW

File: OpenRA.Mods.Common/Scripting/Global/MediaGlobal.cs [hunk 1]
```diff
@@ -188,14 +188,24 @@ public bool PlayMovieInRadar(string movie, LuaFunction playComplete = null)
...
+		public void DisplayMessage(string text, string prefix = "Mission", HSLColor? color = null, int team = 0)
 		{
 			if (string.IsNullOrEmpty(text))
 				return;
 
 			Color c = color.HasValue ? HSLColor.RGBFromHSL(color.Value.H / 255f, color.Value.S / 255f, color.Value.L / 255f) : Color.White;
-			Game.AddChatLine(c, prefix, text);
+			var world = Context.World;
+
+			if (team == 0 || world.LobbyInfo.ClientWithIndex(world.LocalPlayer.ClientIndex).Team == team)
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Scripting/Global/MediaGlobal.cs]: `world.LocalPlayer` will be null during replays. If you used `RenderPlayer`, those messages could be seen in replays as well (with the correct player view selected). You'd still need a null check, though.

File: OpenRA.Mods.Common/Scripting/Global/MediaGlobal.cs [hunk 2]
```diff
@@ -188,14 +188,24 @@ public bool PlayMovieInRadar(string movie, LuaFunction playComplete = null)
...
+				Game.AddChatLine(c, prefix, text);
+		}
+
+		[Desc("Display a text message only to one player.")]
+		public void PrivateMessage(Player player, string text, string prefix = "Mission [Private]", HSLColor? color = null)
+		{
+			if (Context.World.LocalPlayer == player)
```
Comment (obrakmann) [inline OpenRA.Mods.Common/Scripting/Global/MediaGlobal.cs]: `RenderPlayer` might be more nicer here as well.

**Comment by obrakmann** [comment]
Just out of curiosity: what's your use case?

**Comment by pchote** [comment]
RenderPlayer will break normal gameplay with cheats

**Comment by abcdefg30** [comment]
Included a null check for LocalPlayer. Thanks, somehow I completely forgot about that.

> Just out of curiosity: what's your use case?

It was requested for a Coop-mission (so you don't have to spam everyone with only player-specific instructions).

**Comment by obrakmann** [comment]
> RenderPlayer will break normal gameplay with cheats

FWIW, I care more about the game behaving as expected in replays than when the debug options are enabled.

**Comment by abcdefg30** [comment]
We don't show team-messages in replays as well, so I'm not sure if we should here.

**Comment by obrakmann** [comment]
> It was requested for a Coop-mission (so you don't have to spam everyone with only player-specific instructions).

You can already do that in Lua, though, by checking the `IsLocalPlayer` and `Team` properties. That's how the mission objective notifications in our new multiplayer missions work, for example.

**Comment by pchote** [comment]
If we can already do this with existing code then we shouldn't add a new redundant API method.

**Comment by abcdefg30** [comment]
I guess you can. Derp me.

**Pull request closed without merging**